# HG changeset patch
# User Richard Russon <rich@flatcap.org>
# Date 1466119908 -3600
#      Fri Jun 17 00:31:48 2016 +0100
# Branch gentoo-1.7
# Node ID 90f851f933cf6f471a935500554fbf9803a12370
# Parent  3d70abf517990f44165dee996b6bf0e089b43cbc
[PATCH] feature: smime-encrypt-self
From 30402f50427203f349359a50dc0616e4e12f9718 Mon Sep 17 00:00:00 2001
---
 crypt.c | 17 +++++++++++++++--
 init.h  |  6 ++++++
 mutt.h  |  1 +
 3 files changed, 22 insertions(+), 2 deletions(-)


diff -r 3d70abf51799 -r 90f851f933cf crypt.c
--- a/crypt.c	Sat Apr 23 13:33:50 2016 -0700
+++ b/crypt.c	Fri Jun 17 00:31:48 2016 +0100
@@ -231,8 +231,21 @@ int mutt_protect (HEADER *msg, char *key
     if ((WithCrypto & APPLICATION_SMIME)
         && (msg->security & APPLICATION_SMIME))
     {
-      if (!(tmp_pbody = crypt_smime_build_smime_entity (tmp_smime_pbody,
-                                                        keylist)))
+			  char *new_keylist = keylist;
+ 
+			  if (SmimeDefaultKey && query_quadoption(OPT_SMIMEENCRYPTSELF, _("Encrypt message to S/MIME Default Key also?")) == MUTT_YES)
+				  {
+					  /* +1 for NULL, +1 for \n */
+					  int size = mutt_strlen(keylist) + mutt_strlen(SmimeDefaultKey) + 2;
+					  new_keylist = safe_malloc(size);
+					  snprintf(new_keylist, size, "%s%s\n", keylist, SmimeDefaultKey);
+				  }
+ 		 
+			  tmp_pbody = crypt_smime_build_smime_entity (tmp_smime_pbody, new_keylist);
+			  if (new_keylist != keylist)
+				  FREE(&new_keylist);
+			  
+			  if (!tmp_pbody)
       {
 	/* signed ? free it! */
 	return (-1);
diff -r 3d70abf51799 -r 90f851f933cf init.h
--- a/init.h	Sat Apr 23 13:33:50 2016 -0700
+++ b/init.h	Fri Jun 17 00:31:48 2016 +0100
@@ -3285,6 +3285,12 @@ struct option_t MuttVars[] = {
   ** possible \fCprintf(3)\fP-like sequences.
   ** (S/MIME only)
   */
+  { "smime_encrypt_self",		DT_QUAD,	 R_NONE, OPT_SMIMEENCRYPTSELF, MUTT_NO },
+  /*
+  ** .pp
+  ** Encrypt the message to $$smime_default_key too.
+  ** (S/MIME only)
+  */
   { "smime_encrypt_with",	DT_STR,	 R_NONE, UL &SmimeCryptAlg, UL "aes256" },
   /*
   ** .pp
diff -r 3d70abf51799 -r 90f851f933cf mutt.h
--- a/mutt.h	Sat Apr 23 13:33:50 2016 -0700
+++ b/mutt.h	Fri Jun 17 00:31:48 2016 +0100
@@ -299,6 +299,7 @@ enum
   OPT_MIMEFWDREST,
   OPT_MOVE,
   OPT_PGPMIMEAUTO,     /* ask to revert to PGP/MIME when inline fails */
+  OPT_SMIMEENCRYPTSELF,
 #ifdef USE_POP
   OPT_POPDELETE,
   OPT_POPRECONNECT,
